aws/resource_aws_appconfig_application.go

```diff
@@ -0,0 +1,177 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/service/appconfig"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+)
+
+func resourceAwsAppconfigApplication() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsAppconfigApplicationCreate,
+		Read:   resourceAwsAppconfigApplicationRead,
+		Update: resourceAwsAppconfigApplicationUpdate,
+		Delete: resourceAwsAppconfigApplicationDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(1, 64),
+				),
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(0, 1024),
+				),
+			},
+			"tags": tagsSchema(),
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsAppconfigApplicationCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).appconfigconn
+	applicationName := d.Get("name").(string)
+	applicationDescription := d.Get("description").(string)
+
+	input := &appconfig.CreateApplicationInput{
+		Name:        aws.String(applicationName),
+		Description: aws.String(applicationDescription),
+		Tags:        keyvaluetags.New(d.Get("tags").(map[string]interface{})).IgnoreAws().AppconfigTags(),
+	}
+
+	app, err := conn.CreateApplication(input)
+	if err != nil {
+		return fmt.Errorf("Error creating AppConfig application: %s", err)
+	}
+
+	d.SetId(aws.StringValue(app.Id))
```

Comment:
I've been doing this review by comparing the app config code to other examples IE) AwsAccessAnalyzer. 

In the other one, I see it setting it the following way: 
	d.SetId(analyzerName)

Why do we use the ID here? Is the applicationName also a unique identifier? 
What exactly does this line do?

Reply:
SetId sets the resource ID. I'm not sure how access analyzers work, but for AppConfig, AWS generates an ID. The name isn't actually a unique identifier for AppConfig resources. Since we need the resource ID to be unique, the easiest thing to use is the AWS ID. It's also helpful because we need the ID for further API calls.